### DISCUSSION TITLE
multi link namespace

### DISCUSSION THREAD

**Discussion by hjalmarlucius** [geldata/gel#3269] [Ideas]
Thanks for making this fantastic platform with a very well thought through logical structure! One logical concept I miss or haven't found is creating a namespace out of multi link so that the dot notation can be used to traverse multi links, e.g. (disclaimer: it's my second day testing edgeDB so there's surely errors):

```
abstract type Stuff {
  required property name -> str;
  index on (.name);
};
type Pants {
  multi link pocketcontents -> Stuff { KEY_ON .name };  # KEY_ON would be a new keyword
};
select Pants.pocketcontents.nameA;
```

Reasons I like it:
* it is a nice shorthand alternative to a query
* it is a structural pattern that I very frequently use
* it carries well over into a e.g. a python `dict[str, Stuff]`
* it maintains typing consistency

**Comment by elprans** [comment]
It's an interesting idea for dictionary-like objects.  If implemented, it would likely have to use some syntax other than path navigation, e.g `Pants.pocketcontents['nameA']` or something like that.  Primarily because of non-string keys, but also to avoid clashing with path syntax.

**Comment by hjalmarlucius** [comment]
Yes agree with that, this would also avoid muddling the distinction between schema and data.
